src/openzaak/utils/middleware.py

```diff
@@ -15,3 +33,72 @@ def __call__(self, request):
 
     def log(self, request: HttpRequest):
         logger.debug("Request headers for %s: %r", request.path, request.headers)
+
+
+class JWTAuth(_JWTAuth):
+    def __init__(self, encoded: str = None):
+        self.encoded = encoded
+        self.component = None
+
+    @property
+    def applicaties(self) -> Union[list, None]:
+        if self.client_id is None:
+            return None
+
+        return Applicatie.objects.filter(client_ids__contains=[self.client_id])
+
+    def get_autorisaties(self, init_component: str) -> models.QuerySet:
+        """
+        Retrieve all authorizations relevant to this component.
+        """
+        component = COMPONENT_MAPPING.get(init_component, init_component)
+        app_ids = self.applicaties.values("id")
+        return Autorisatie.objects.filter(
+            applicatie_id__in=Subquery(app_ids), component=component
+        )
+
+    def has_auth(self, scopes: List[str], init_component: str = None, **fields) -> bool:
+        if scopes is None:
+            return False
+
+        if not self.applicaties:
+            return False
+
+        # allow everything
+        if self.applicaties.filter(heeft_alle_autorisaties=True).exists():
+            return True
+
+        if not init_component:
+            return False
+
+        autorisaties = self.get_autorisaties(init_component)
+        scopes_provided = set()
+
+        # filter on all additional components
+        for field_name, field_value in fields.items():
+            if hasattr(self, f"filter_{field_name}"):
+                autorisaties = getattr(self, f"filter_{field_name}")(
+                    autorisaties, field_value
+                )
+            else:
+                autorisaties = self.filter_default(
+                    autorisaties, field_name, field_value
+                )
+
+        for autorisatie in autorisaties:
+            scopes_provided.update(autorisatie.scopes)
+
+        return scopes.is_contained_in(list(scopes_provided))
+
+
+class AuthMiddleware(_AuthMiddleware):
+    def extract_jwt_payload(self, request):
+        authorization = request.META.get(self.header, "")
+        prefix = f"{self.auth_type} "
+        if authorization.startswith(prefix):
```

Comment:
I'm not sure what's happening here?

Reply:
removing "Bearer " from the "Authorization" header value I suppose

Reply:
oooh of course